### PR TITLE
Add parsing support to := operator

### PR DESCRIPTION
**Pull request by Ayman161803**
## Description
This PR adds support to assignment operator `:=` to parse queries of the type mentioned in #10760.
The MySQL docs can be found here - https://dev.mysql.com/doc/refman/8.0/en/assignment-operators.html

## Related Issue(s)
#10760 
#8604 

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported
-   [ ] Tests were added or are not required
-   [ ] Did the new or modified tests pass consistently locally and on the CI
-   [ ] Documentation was added or is not required

### CODE REVIEW

**Comment by GuptaManan100** [comment]
@vitessio/query-serving This is good from @Ayman161803. The current changes adds the parsing support for the assignment operator, but I tested out our planner, it doesn't handle it well. The planner needs to realize that an assigment operator is basically assigning the user defined variable and then using it, so we need to plan it as a set and then whatever the command is.

**Comment by GuptaManan100** [comment]
I have explicitly disallowed this expression in the planner for now.

File: go/vt/sqlparser/parse_test.go [hunk 1]
```diff
@@ -1234,6 +1234,8 @@ var (
 		input: "update /* simple */ a set b = 3",
 	}, {
 		input: "update /* a.b */ a.b set b = 3",
+	}, {
+		input: "update a.b set d = @v := d + 7 where u = 42",
```
Comment (harshit-gangal) [inline go/vt/sqlparser/parse_test.go:1238]: add a select case as well
```mysql
SELECT @var1:=COUNT(*) FROM t1;
```

File: go/vt/vtgate/planbuilder/testdata/unsupported_cases.json [hunk 2]
```diff
@@ -467,5 +467,30 @@
     "comment": "mix lock with other expr",
     "query": "select get_lock('xyz', 10), 1 from dual",
     "plan": "VT12001: unsupported: LOCK function and other expression: [1] in same select query"
+  },
+  {
+    "comment": "Assignment expression in select statements",
+    "query": "select @val := 42",
+    "plan": "VT12001: unsupported: Assignment expression"
+  },
```
Comment (GuptaManan100) [inline go/vt/vtgate/planbuilder/testdata/unsupported_cases.json:475]: @harshit-gangal There is one.

Comment (harshit-gangal) [reply go/vt/vtgate/planbuilder/testdata/unsupported_cases.json:475]: I was looking for it in the parse test.

**Review by harshit-gangal — APPROVED** [review]

**Review by GuptaManan100 — APPROVED** [review]

**Pull request merged**